src/node/mini_miner.cpp

```diff
@@ -292,7 +293,11 @@ void MiniMiner::BuildMockTemplate(const CFeeRate& target_feerate)
         DeleteAncestorPackage(ancestors);
         SanityCheck();
     }
-    Assume(m_in_block.empty() || m_total_fees >= target_feerate.GetFee(m_total_vsize));
+    if (!target_feerate.has_value()) {
+        Assume(m_in_block.size() == num_txns);
```

Comment:
It took me until the commit `[MiniMiner] track inclusion order and add Linearize() function` until I understood what this PR was aiming to achieve. Perhaps you could add another sentence to the PR description about the goal of this PR.